README.md

```diff
@@ -1,3 +1,339 @@
+
+#### `--pre-release`
+To allow for installation of pre-release versions
+
+#### `--with-php-config=/path/to/php-config`
+
+Allows installation of extensions with PHP versions that are not in the path
+
+#### `--{option}{?=value}`
+
+All options specified in the `config` section of the composer.json file can also be given, including a value if they
+take them. For example, for Xdebug you could run `npecl install xdebug --without-xdebug-compression`.
+
+## Extension maintainer: register an npecl package
+
+Create a `composer.json` in your repository, commit:
+
+```json
+{
+    "name": "ext-xdebug",
+    "type": "php-ext-zend",
+    "license": "Xdebug-1.03",
+    "description": "Xdebug is an amazing tool...",
+    "require": {
+        "ext-zlib": "*",
+        "php": ">=7.4.21,<8.4"
+    },
+    "conflict": {
+        "ext-zetalib": "*"
+    },
+    "php-ext": {
+        "priority": 90,
+        "config": {
+            "enable-xdebug-dev": false,
+            "without-xdebug-compression": false
+        }
+    }
+}
+```
+
+Submit your repo URL into [https://packagist.org/packages/submit](https://packagist.org/packages/submit). Packagist
+will read the `composer.json` as usual.
+
+When releases are made, Packagist will do the usual thing of producing metadata for the release
+
+### Windows binaries
+
+Windows needs pre-built binary DLLs. The expected workflow is that the release is made, then some kind of build takes
+place (for example, in a GitHub Action, or manually on a compatible build environment), and the `.dll` file is added to
+the GitHub release.
+
+The name for the DLLs must follow the following pattern:
+
+ * `php_{extension-name}-{tag}-{php-maj/min}-{compiler}{"-nts"}?{-arch}?.dll`
+
+The descriptions of these items:
+
+ * `extension-name` the name of the extension, e.g. `xdebug` - this is the same as defined in `composer.json` minus `ext-`
+ * `tag` for example `3.3.0alpha3` - defined by the tag/release you have made
+ * `php-maj/min` - for example `8.3` for PHP 8.3.*
+ * `compiler` - usually `vc16`?
+ * `-nts` - optional - non-thread safe. If omitted, thread-safe mode is assumed
+ * `-arch` - optional - for example `x86_64`, fetch using `uname -i`. If omitted, the extension is assumed to be
+   compatible with all architectures
+
+### Notes on the `composer.json`
+
+ * The `name` is not a typical `vendor/package` format, it is `ext-package` format.
+ * The `type` dictates if it is a PHP Module (`php-ext`) or a Zend Extension (`php-ext-zend`).
+ * Typically, there will be almost no `require` definitions, except `php` version itself
+ * The `php-ext` is a new top-level element to provide additional metadata for building the extension, if required.
+   * `priority` can be used to prefix to the INI file, e.g. `90-xdebug.ini` which affects the loading order. The
+     priority is a number in the range 10-99 inclusive. There are two digits so that the files sort correctly on any
```

Comment:
priority is confusing (especially as a higher value will mean loaded last)

As it affects load "load order", better to call it **load_order**.

Reply:
I don't agree here. It *is* a priority. The later the extension is loaded (higher priority) the later it sets up its hooks, which means it has *priority* access to those being called.

Also, `load_order` or `loadOrder`? It's the clunkier name.

Reply:
> The later the extension is loaded (higher priority) the later it sets up its hooks, which means it has priority access to those being called.

It would be good to add a similar explanation. It describes straightforwardly the intent.

Reply:
Clarified this in the latest update, thanks folks. I think priority is ok, as long as we have it defined clearly in docs; it is optional anyway